src/d3a/models/area/event_dispatcher.py

```diff
@@ -126,19 +126,46 @@ def event_listener(self, event_type: Union[MarketEvent, AreaEvent], **kwargs):
             self.area.strategy.event_on_disabled_area()
 
     @staticmethod
-    def select_agent_class(is_spot_market):
+    def create_agent_object(owner, higher_market, lower_market, is_spot_market):
         if is_spot_market:
             if ConstSettings.IAASettings.MARKET_TYPE == 1:
                 if ConstSettings.IAASettings.AlternativePricing.PRICING_SCHEME != 0:
-                    return OneSidedAlternativePricingAgent
+                    return OneSidedAlternativePricingAgent(
+                        owner=owner,
+                        higher_market=higher_market,
+                        lower_market=lower_market,
+                        min_offer_age=ConstSettings.IAASettings.MIN_OFFER_AGE
+                    )
                 else:
-                    return OneSidedAgent
+                    return OneSidedAgent(
+                        owner=owner,
+                        higher_market=higher_market,
+                        lower_market=lower_market,
+                        min_offer_age=ConstSettings.IAASettings.MIN_OFFER_AGE
+                    )
```

Comment:
```suggestion
                return agent_class(owner=owner, higher_market=higher_market,
                                   lower_market=lower_market,
                                   min_offer_age=ConstSettings.IAASettings.MIN_OFFER_AGE)

```

Reply:
I see, so you want to avoid duplication in assignment of constructor arguments here, good idea. I'll do it, thanks